rest/main.go

```diff
@@ -102,12 +108,28 @@ func serverMainPersistentConfig(fs *flag.FlagSet, flagStartupConfig *StartupConf
 	if len(configPath) == 1 {
 		fileStartupConfig, err := LoadStartupConfigFromPath(configPath[0])
 		if pkgerrors.Cause(err) == base.ErrUnknownField {
-			// TODO: CBG-1399 Do automatic legacy config upgrade here
-			base.Warnf("Couldn't parse bootstrap config and legacy config migration not yet implemented: %v", err)
-			// When automatic legacy config upgrade is done return when disable_persistent_config=true
-			return true, nil
-		}
-		if err != nil {
+			// If we have an unknown field error processing config its possible that the config is a 2.x config
+			// requiring automatic upgrade. We should attempt to perform this upgrade
+
+			var upgradeError error
+			fileStartupConfig, disablePersistentConfigFallback, upgradeError = automaticConfigUpgrade(configPath[0])
+			if upgradeError != nil {
+
+				// We need to validate if the error was again, an unknown field error. If this is the case its possible
+				// the config is actually a 3.x config but with a genuine unknown field, therefore we should  return the
+				// original error from LoadStartupConfigFromPath.
+				if pkgerrors.Cause(upgradeError) == base.ErrUnknownField {
+					return false, err
+				}
+
+				return false, upgradeError
+			}
+
+			if disablePersistentConfigFallback {
+				return true, nil
+			}
+
```

Comment:
There is no logging around any of this. It makes it really difficult to tell what's happening on startup/migration/upgrade. Given it's a rather disruptive change, it would make sense to be pretty verbose about what we're doing (e.g. "backing up 2.x config to '...'")

```
> go build && ./sync_gateway cnf_dbg.json
2021-08-03T20:01:32.031+01:00 ==== Couchbase Sync Gateway/() CE ====
2021-08-03T20:01:32.031+01:00 [INF] Loading content from [cnf_dbg.json] ...
2021-08-03T20:01:32.031+01:00 [INF] Loading content from [cnf_dbg.json] ...
2021-08-03T20:01:33.267+01:00 [INF] Config: Starting in persistent mode using config group "default"
2021-08-03T20:01:33.267+01:00 [INF] Logging: Console to stderr
...
```